src/calculations.js

```diff
@@ -24,23 +24,32 @@ const calculateNewPurchaseValues = data => {
   let today = new Date();
   let now = dayjs(today);
 
-  let lastPurchaseDate =
-    data.dateOfPurchase instanceof Date
-      ? data.dateOfPurchase
-      : data.dateOfPurchase.toDate();
-  let latestInterval = now.diff(dayjs(lastPurchaseDate), 'day');
-  let newEstimate = calculateEstimate(
-    data.numberOfDays,
-    latestInterval,
-    data.numberOfPurchases,
-  );
+  if (data.dateOfPurchase === undefined) {
+    return {
+      id: data.id,
+      numberOfDays: data.numberOfDays,
+      numberOfPurchases: data.numberOfPurchases + 1,
+      dateOfPurchase: today,
+    };
+  } else {
```

Comment:
Also, you don't need the `else` here because if the users gets into `if` the function will return and will be impossible to reach anything after that (`return` statement)